### PR TITLE
Update mandarin branch with more features

### PR DESCRIPTION
**Pull request by yanganto**
* add translate option
  - with `--trans` or `-t` the traditional Chinese word will translate into simplified Chinese
  - default will not translate

* add suitable length feature, aux symbols, fix test
  - add the longest length feature
  - add the shortest length feature
  - add patch aux symbol feature

* fix test case

* update app info
  - add maintainer and email

### CODE REVIEW

**Comment by yanganto** [comment]
Hi here, this commit is update the mandarin branch based on some feature requests from common voice in Hong Kong telegram group.  If this PR is suitable please kindly keep this.  Thanks.

File: src/app.rs [hunk 1]
```diff
@@ -30,6 +30,33 @@ where
                         .takes_value(true)
                         .number_of_values(1)
                         .help("input dir to glob"),
+                )
+                .arg(Arg::with_name("trans").short("t").long("trans").help(
```
Comment (MichaelKohler) [inline src/app.rs:34]: minor issue: can you put those on separate lines as well?

File: src/app.rs [hunk 2]
```diff
@@ -30,6 +30,33 @@ where
...
+                    "automatically translate words from traditional Chinese into simplify Chinese",
```
Comment (MichaelKohler) [inline src/app.rs:35]: minor: *simplified

File: src/app.rs [hunk 3]
```diff
@@ -51,14 +78,36 @@ where
...
+        .collect();
 
     let mut sentences = vec![];
     for file_name in file_names {
         eprintln!("file_name = {:?}", file_name.to_string_lossy());
         let texts = load(&file_name)?;
         for text in texts {
             let mut article_sentences = vec![];
-            for next in SentenceExtractor::new(&text) {
+            for next in SentenceExtractor::new_with_opt(
```
Comment (MichaelKohler) [inline src/app.rs:104]: I don't think we need to rename that, what do you think?

File: src/extractor.rs [hunk 4]
```diff
@@ -68,57 +113,34 @@ impl Iterator for SentenceExtractor {
...
                 .collect::<String>();
 
+            // remove words in brackets
             next_item = PARANS.replace(&next_item, "").to_string();
-            next_item = next_item
-                .chars()
-                .map(|c| CHARACTER_MAP.get(&c).unwrap_or(&c).clone())
-                .collect();
 
+            // transalte words into simplify format
```
Comment (MichaelKohler) [inline src/extractor.rs:119]: minor: *translate, *simplified

**Comment by yanganto** [comment]
Thanks for the review.  

The straightforward part base on your advice is amended.  

I am not directly working on this project but willing to fix issues and provided features on this tool, such that some questions aforementioned are out of my capability, such that I can not make a response on these.  Please kindly understand.

I will bring your suggestion back.  
Thanks.

**Comment by MichaelKohler** [comment]
Merged into the new zh-hk branch

**Pull request merged**